lib/mongodb-connect.js

```diff
@@ -0,0 +1,55 @@
+const mongo = require('mongodb');
+const dns = require('dns');
+
+// Connect to MongoDB, using the modern topology and parser, and
+// a tolerant policy to successfully connect to "localhost" even if
+// the first record returned by the resolver doesn't reach mongodb's
+// bind address because localhost resolves first to ::1 (ipv6) and
+// mongodb lists only on 127.0.0.1 (ipv4) by default. For broadest
+// compatibility we don't assume we know this will happen, we try all the
+// addresses that localhost actually resolves to and succeed with the
+// first one that works.
+
+module.exports = async (uri, options) => {
+  const connectOptions = {
+    useUnifiedTopology: true,
+    useNewUrlParser: true,
+    ...options
+  };
+  const parsed = new URL(uri);
+  if ((parsed.protocol !== 'mongodb:') || (parsed.hostname !== 'localhost')) {
+    return mongo.MongoClient.connect(parsed.toString(), connectOptions);
+  }
+  const records = await dns.promises.lookup('localhost', { all: true });
+  if (!records.length) {
+    // The computer that reaches this point has bigger problems 😅
+    throw new Error('Unable to resolve localhost to an IP address.');
+  }
+  return new Promise((resolve, reject) => {
```

Comment:
I can't use `Promise.all` because I need to succeed if any of them work and I need to clean up in a specific way if they don't. So there's custom logic here to resolve and reject.

Reply:
What about `Promise.any()`?

> This method is useful for returning the first promise that fulfills
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any

Reply:
...and it's only available on browsers